STYLE: Change template parameter name prefix N to V

Leengit opened the pull request:
Template parameters that are of `typename` have names that start with `T`, but for template parameters that are values (e.g., `unsigned int`) rather than types, both `N` and `V` have been used as a prefix for their names.  This pull request changes instances with prefix `N` to have a `V` prefix.  Additionally `NDimensions` (plural) is changed to `VDimension` (singular).

Note that class members (typically `static const(expr)` members) are sometimes named with an `N` or `V` prefix.  Perhaps neither of these prefixes is appropriate by the ITK Style Guide.  Nonetheless, these are not changed because outside code, such as ITK modules, might refer to them by their current names. 

Thanks to @N-Dekker for discovering this issue.

## PR Checklist
- [X] No [API changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#breaking-changes) were made (or the changes have been approved)
- [X] No [major design changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#design-changes) were made (or the changes have been approved)
- [X] Added test (or behavior not changed)
- [X] Updated API documentation (or API not changed)
- [X] Added [license](https://github.com/InsightSoftwareConsortium/ITK/blob/master/Utilities/KWStyle/ITKHeader.h) to new files (if any)
- [X] Added Python wrapping to new files (if any) as described in [ITK Software Guide](https://itk.org/ItkSoftwareGuide.pdf) Section 9.5
- [X] Added [ITK examples](https://github.com/InsightSoftwareConsortium/ITKSphinxExamples) for all new major features (if any)

N-Dekker commented:
Great job @Leengit 👍  Thanks for addressing the issue that I raised at discussion topic [[Which initial is preferred for a non-type template parameter, N or V (`NDimension` or `VDimension`)?](https://discourse.itk.org/t/which-initial-is-preferred-for-a-non-type-template-parameter-n-or-v-ndimension-or-vdimension/4788)

Do you think we should then also adjust [Utilities/KWStyle/ITK.kws.xml](https://github.com/InsightSoftwareConsortium/ITK/blob/master/Utilities/KWStyle/ITK.kws.xml)? I don't really know how that works, but the xml code `<Template>([TNV]...` seems to allow both `N` and `V`.

Some minor comments regarding your commit messages:

    BUG: Revert b04d9cd for member variables.

I'm not sure if that particular SHA (b04d9cd) will make it onto the master branch. During rebase/merge, the SHA of the commit may change. It's probably more safe to just say something like:

    BUG: Revert changes from previous commit for member variables

But I guess you might as well squash both commits to one, if that's OK to you.

I noticed that you add a period ('.') to the subject line of commit messages. Is that really helpful to you? Because it seems a bit odd to me. A large majority of commit messages certainly don't have that period. And it does not seem necessary to me, because it is just the _subject_, not necessarily a complete English sentence. And as the number of characters preferred for a subject line is limited (maximum 72 character), I wouldn't want to start adding periods to my subject lines!

Leengit commented:
Thank you @N-Dekker for the suggestions.  With this force-push, I have merged the commits, removed the SHA code and trailing period from the commit message, and removed `N` from [Utilities/KWStyle/ITK.kws.xml](https://github.com/InsightSoftwareConsortium/ITK/blob/990aaf94cb690308251aecbfcbf5459dd2cac560/Utilities/KWStyle/ITK.kws.xml).

N-Dekker commented on Modules/Video/BridgeOpenCV/test/itkOpenCVBasicTypeBridgeTest.cxx:
```diff
@@ -169,16 +169,16 @@ Size2ConversionTest()
-  constexpr unsigned int NColumns = 3;
-  using ITKMatrixType = itk::Matrix<double, NRows, NColumns>;
-  using CVMatrixType = cv::Matx<double, NRows, NColumns>;
+  constexpr unsigned int VRows = 2;
+  constexpr unsigned int VColumns = 3;
```
`VRows`, `VColumns`? Maybe it should be `NumberOfRows`, `NumberOfColumns`, in this case? These are not formal template parameters anyway... 🤔

Leengit replied:
Those aren't template parameters, and if they were in non-test code then they shouldn't have been changed at all.  I will change them to avoid the `V` and `N` conventions entirely (well, sort of): `NumberOfRows` and `NumberOfColumns`

N-Dekker commented on Modules/Video/BridgeOpenCV/include/itkOpenCVBasicTypeBridge.h:
```diff
@@ -167,11 +167,11 @@ struct OpenCVBasicTypeBridge<itk::Size<2>, cv::Size>
 
-template <typename T, unsigned int NRows, unsigned int NColumns>
-struct OpenCVBasicTypeBridge<itk::Matrix<T, NRows, NColumns>, cv::Matx<T, NRows, NColumns>>
+template <typename T, unsigned int VRows, unsigned int VColumns>
```
In this case, the old `NRows` and `NColumns` already suggested that these are the _number of_ rows and columns, respectively. When using the `V` initial instead, we might need to make it a little bit clearer, by naming them `VNumberOfRows`, `VNumberOfColumns`. What do you think?

Leengit replied:
I thought that the old `NRows`meant "non-type rows" or "numeric rows" rather than "number of rows".  I am thinking to leave `VRows` as is.

N-Dekker replied:
OK!

dzenanz reviewed (approved):
Looks good on a glance. There were also quite some manual interventions here!

Leengit commented:
I'd like to see how this does with overnight testing runs because it touches a lot of files.  If all remains green then this should be good to merge.

N-Dekker reviewed (approved):
Approved, even though personally I would have preferred consistent use of the `N`, rather than the `V` as prefix for non-type template parameters.

Leengit commented:
Rebased.  Will add to `.git-blame-ignore-revs` after merging.

Leengit commented:
This past all its tests.  I have squashed, rebased to current master, and force-pushed.  This is ready for merging.

Leengit commented:
I rebased against the newly current master and force-pushed.  The strange message from kwrobot-v1 makes it seem like my force-push collided with a merge attempt by someone else.  If that's what is going on, attempting the merge again might be the right course of action.  If not, please help me to understand the message from kwrobot-v1.

dzenanz commented:
The KWRobot message came 1 second after my approval, so I think that is what caused it. Your PR is not merged.

Leengit commented:
> The KWRobot message came 1 second after my approval, so I think that is what caused it. Your PR is not merged.

Probably then my force-push collided with your approval; perhaps KWRobot was trying to say that you were approving a commit that had been obsoleted in the seconds before-hand.

Perhaps extra caution is appropriate nonetheless.  We might wait for the checks to pass (again) before merging.

dzenanz commented:
Your force-push was 5 minutes before my approval. So no, I can't imagine those two colliding.

dzenanz commented:
Than that was probably it. I now see that force-push, but I didn't see it when I wrote https://github.com/InsightSoftwareConsortium/ITK/pull/3171#issuecomment-1036271137.

Leengit commented:
Because one of the checks was stalled for several days, I just took the opportunity to rebase to the current master branch and force push.  There are no other changes.

dzenanz commented:
This is probably what broke FEM module. It is off by default. I have it enabled locally, for testing. I now run into:
```text
96>------ Build started: Project: ITKFEMRegistration, Configuration: Debug x64 ------
96>itkFEMRegistrationFilter.cxx
96>C:\Dev\ITK-git\Modules\Numerics\FEM\include\itkImageToRectilinearFEMObjectFilter.h(72,54): error C2065: 'VDimension': undeclared identifier
96>C:\Dev\ITK-git\Modules\Numerics\FEM\include\itkImageToRectilinearFEMObjectFilter.h(175): message : see reference to class template instantiation 'itk::fem::ImageToRectilinearFEMObjectFilter<TInputImage>' being compiled
96>C:\Dev\ITK-git\Modules\Numerics\FEM\include\itkImageToRectilinearFEMObjectFilter.h(72,1): error C2975: 'VDimension': invalid template argument for 'itk::fem::FEMObject', expected compile-time constant expression
96>C:\Dev\ITK-git\Modules\Numerics\FEM\include\itkFEMObject.h(74): message : see declaration of 'VDimension'
95>ITKIOHDF5.vcxproj -> C:\Dev\ITK-vs19\lib\Debug\ITKIOHDF5-5.3.lib
96>Done building project "ITKFEMRegistration.vcxproj" -- FAILED.
```

dzenanz commented on Modules/Numerics/FEM/include/itkImageToRectilinearFEMObjectFilter.h:
```diff
@@ -69,7 +69,7 @@ class ITK_TEMPLATE_EXPORT ImageToRectilinearFEMObjectFilter : public ProcessObje
 
   /** Typedefs for Output FEMObject */
-  using FEMObjectType = typename itk::fem::FEMObject<NDimensions>;
+  using FEMObjectType = typename itk::fem::FEMObject<VDimension>;
```
Definition of this on line 60 was not updated.

dzenanz replied:
Addressed in #3201.
